src/server/ability.ts

```diff
@@ -0,0 +1,25 @@
+import {AbilityBuilder, Ability} from '@casl/ability'
+import {loadCurrentUser} from 'lib/users'
+
+type Actions = 'manage' | 'upload'
+type Subjects = 'Video' | 'all'
+type AppAbility = Ability<[Actions, Subjects]>
+
+export async function getAbilityFromToken(token: string) {
+  const user = await loadCurrentUser(token)
+  return defineAbilityFor(user)
+}
+
+function defineAbilityFor(user: any) {
+  const {can, build} = new AbilityBuilder<AppAbility>(Ability)
```

Comment:
```suggestion
  // AbilityBuilder https://casl.js.org/v5/en/guide/define-rules#ability-builder-class
  const {can, build} = new AbilityBuilder<AppAbility>(Ability)
```

Reply:
The `AbilityBuilder` API is a little indirect, so I thought it'd be handy to link to the docs right above it.